http.go

```diff
@@ -92,7 +103,7 @@ func (client *Client) Request(rawPayload []byte, method, route string, query url
 				return nil, nil, err
 			}
 		}
-		resp, err := (&http.Client{}).Do(callSummary.HTTPRequest)
+		resp, err := DefaultHTTPClient.Do(callSummary.HTTPRequest)
```

Comment:
I'm with @walac on this one, in my opinion we should work with an interface, even if a single method is defined (for now).

Regarding the `DefaultHTTPClient` pattern, it is something that is common throughout a lot of go code that I've read (even [net/http](https://golang.org/src/net/http/client.go#L110) itself does it). However, if we agree to instead always use `http.Client`, I'll make the change.